ffigen.schema.json

```diff
@@ -0,0 +1,404 @@
+{
+    "$id": "https://json.schemastore.org/ffigen",
+    "$schema": "https://json-schema.org/draft/2020-12/schema",
+    "description": "Configuration for generating dart ffi bindings via ffigen",
+    "type": "object",
```

Comment:
Can this file be generated from our configuration spec in Dart?

The benefits:

1. We can hit generate when we add config logic.
2. Either all properties are working or wrong, making it less likely to have typos in this spec.
3. We can generate the schema on the CI, checking it's up to date.